app/models/annotation_category.rb

```diff
@@ -61,6 +61,7 @@ def self.add_by_row(row, assignment, current_user)
       annotation_category.update!(flexible_criterion_id: criterion.id)
       row.each_slice(2) do |text_with_deduction|
         begin
+          raise ArgumentError if text_with_deduction.second.nil?
```

Comment:
Instead of doing this, catch rescue a `TypeError` in addition to `ArgumentError`.